### PR TITLE
Bigtable Data client: mutations apis

### PR DESCRIPTION
**Pull request by jiren**
- Single mutate single row, mutates multiple rows in a batch, read modify and write row
- Unit test cases using mocks

### CODE REVIEW

**Comment by jiren** [comment]
> This looks good structurally. I didn't review anything related to the design. eg. Table vs. Client::Table

I am going to remove `Bigtable::Client::Table` (class for table data operations) and going to merge into `Bigtable::Table` class.

File: google-cloud-bigtable/test/google/cloud/bigtable/client/table_test.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+describe Google::Cloud::Bigtable::Client::Table, :table, :mock_bigtable do
+  let(:instance_id) { "test-instance-id" }
+  let(:table){
+    Google::Cloud::Bigtable::Client::Table.new(Object.new, "dummy-table-path")
+  }
+
+  it "create mutation entry instance" do
+    mutation_entry = table.new_mutation_entry("row-1")
+    mutation_entry.must_be_kind_of Google::Cloud::Bigtable::MutationEntry
```
Comment (quartzmo) [inline google-cloud-bigtable/test/google/cloud/bigtable/client/table_test.rb]: Add line with assertion for `row_key`?

File: google-cloud-bigtable/test/google/cloud/bigtable/client/table_test.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  it "create read modify write row rule instance" do
+    rule = table.new_read_modify_write_rule("cf", "field1")
+    rule.must_be_kind_of Google::Cloud::Bigtable::ReadModifyWriteRule
```
Comment (quartzmo) [inline google-cloud-bigtable/test/google/cloud/bigtable/client/table_test.rb]: Add lines with assertions for `family` and `qualifier`?

**Comment by blowmage** [comment]
@jiren This PR is assigned to you to merge.

**Comment by sduskis** [comment]
@jiren, do you have the ability to merge this PR?

**Comment by jiren** [comment]
@blowmage and @quartzmo, If you are ok with changes please merge this pull request.

**Pull request merged**